File: 02_CSharpPart2/Methods-Homework/04.AppearanceCount/AppearanceCount.cs
/*
 * Problem 4. Appearance count
 * Write a method that counts how many times given number appears in given array.
 * Write a test program to check if the method is workings correctly.
*/
using System;
using System.Linq;
class AppearanceCount
{
    static void NumberCount (int[] array, int number)
    {
        int counter = 0;
        for (int i = 0; i < array.Length; i++)
        {
            if (number == array[i])
            {
                counter++;
            }
        }
        Console.WriteLine("Number {0} appears {1} times in the array", number, counter);

    }
    static void Main()
    {
        Console.WriteLine("Enter N numbers separated by space or coma to the Array: ");
        int[] array = Console.ReadLine()
           .Split(new char[] { ' ', ',' }, StringSplitOptions.RemoveEmptyEntries)
           .Select(x => int.Parse(x))
           .ToArray();

        Console.Write("For which number are you looking: ");
        int number = int.Parse(Console.ReadLine());

        NumberCount(array, number);
    }
}

File: 06_JavaScript-Fundamentals/Functions-Homework/05.AppearanceCount/AppearanceCount.js
/* Problem 5. Appearance count

Write a function that counts how many times given number appears in given array.
Write a test function to check if the function is working correctly.
*/
var array = [],
    loops = 20, // You may edit the length of the array
    number,
    counter,
    i;

testTask(loops);

function testTask(loops) {

    array = randomNumbers(loops); // Random generated array with numbers between 1 and 5
    number = array[Math.random() * loops | 0]; // Picking a random position out of the array
    
    console.log('Initial array is: ' + array.join(', '));
    console.log('You can find the number ' + number + ' in this array ' + findOccurrence(array, number) + ' times');
}

function findOccurrence (array, number) {

    counter = 0;
    loops = array.length;

    for (i = 0; i < loops; i += 1) {

        if (array[i] === number) {
            counter += 1;
        }
    }

    return counter;
}

function randomNumbers(loops) {

    for (i = 0; i < loops; i += 1) {

        array.push((Math.random() * 5 + 1 ) | 0);
    }

    return array;
}